src/patch_parser_test.py

```diff
@@ -70,16 +71,137 @@ def test_parse_with_replies(self):
                     message='On Mon, 31 Aug 2020 at 12:04:46 +0100, The Sender wrote:'),
             Comment(raw_line=18,
                     file='file',
-                    line=7, # TODO: should be 5
+                    line=7,  # TODO: should be 5
                     message='Comment on old line 5, want on line 5 in new file.'),
             Comment(raw_line=20,
                     file='file',
                     line=9,  # TODO: should be 7
                     message='Comment on old line 7, want on line 8 in new file.'),
         ])
 
+    def test_one_modified_line(self):
+        raw_patch = '''
+            Commit message goes here.
+
+            ---
+             file | 3 ++-
+             1 file changed, 2 insertions(+), 1 deletion(-)
+
+            diff --git a/file b/file
+            index fa2da6e55caa..1fc93eb38351 100644
+            --- a/file
+            +++ b/file
+            @@ -2,7 +2,8 @@ line 1
+             line 2  # this is line 11 in raw_patch
+             line 3
+             - line 4
+             + line 4 - edit
+             line 5
+             line 6
+             line 7  # this is line 16 in raw_patch
+
+            base-commit: 235360eb7cd778d7264c5e57358a3d144936b862
+            --
+                    '''.strip()
+        line_map = patch_parser._parse_git_patch(raw_patch)
+        self.assertEqual(line_map.map(11), ('file', 2), msg=repr(line_map))  # line 2, control assert
+        self.assertEqual(line_map.map(13), ('fileb', 4), msg=repr(line_map))  # line 4 (deleted part due to edition)
+        self.assertEqual(line_map.map(14), ('file', 5), msg=repr(line_map))  # line 4 the modified line
+        self.assertEqual(line_map.map(16), ('file', 7), msg=repr(line_map))  # line 7, control assert
+
+
+    def test_one_deleted_line(self):
```

Comment:
I like this, yeah we should have had simpler cases like this.
Really helps isolate the issues, thanks!

Reply:
+1 Nice job!